File: backend/src/main/java/com/example/model/BasicNote.java
package com.example.model;

import lombok.Data;
import lombok.Getter;
import lombok.Setter;

import javax.persistence.Embeddable;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;

@Entity
public class BasicNote extends Note{

    @Getter
    @Setter
    String front;

    @Getter
    @Setter
    String back;

    BasicNote(){}

    public BasicNote(String front, String back){
        this.front = front;
        this.back = back;
    }

    @Override
    public String toString() {
        return "Basic Note {front:" + front + ", back:" + back + "}";
    }
}

File: src/types/BasicNote.ts
import Note from 'Note';

export default class BasicNote extends Note{

    private _front: string;
    private _back: string;

    constructor(front: string, back: string, id: string) {
        super(id);
        this._front=front;
        this._back=back;
    }

    get front(): string {
        return this._front;
    }

    set front(front: string) {
        this._front = front;
    }

    get back(): string {
        return this._back;
    }

    set back(back: string) {
        this._back = back;
    }
}